visualeyes/core/plotting.py

```diff
@@ -1,99 +1,79 @@
+        epoch_x = x_coord[x_coord['epoch_index'] == index]
+        epoch_y = y_coord[y_coord['epoch_index'] == index]
 
-        # locate the fixations within trial start and end time
-        fixations_in_trial = df_fixations[(df_fixations['stime'] >= trial_stime) & (df_fixations['etime'] <= trial_etime)]
+        out_of_bounds = any(x < 0 or x >= aoi_mask.shape[1] or y < 0 or y >= aoi_mask.shape[0]
+                           for x,y in zip(epoch_x, epoch_y)
+                           )
 
-        # find the x and y coordinates of fixations 
-        x_coor = fixations_in_trial ['axp']
-        y_coor = fixations_in_trial ['ayp']
+        if out_of_bounds:
+            print(f"Some coordinates for epoch {index} are out of bounds")
 
-        # count the number of fixations
-        num_fixations = x_coor.shape[0]
+        else:
 
-        # generate the binary image as aoi mask
-        edf_file = read_edf(filepath)
-        
-        # extract screen coordinates from the 'info' column of the edf file
-        screen_width = edf_file['info']['screen_coords'][0]
-        screen_height = edf_file['info']['screen_coords'][1]
-
-        # define boundaries of the aoi (assume rectangle shape)
-        SHAPE = 'rectangle'
-        PIXELS = 100
-        x1 = screen_width/2 - PIXELS
-        x2 = screen_width/2 + PIXELS
-        y1 = screen_height/2 - PIXELS
-        y2 = screen_height/2 + PIXELS
-        
-        # initiate aoi parameters dict
-        aoi_parameters = [{'shape': SHAPE, 'coordinates': (x1, y1, x2, y2)}]
+            plt.imshow(aoi_mask, cmap='gray')
+
+            plt.title(f"plot epoch {index} data on AOI mask")
+
+            plt.set_xlim(0, aoi_mask.shape[1]-1)
 
-        # generate aoi mask
-        aoi_mask = define_aoi(screen_width, screen_height, aoi_parameters)
+            plt.set_ylim(0, aoi_mask.shape[0]-1)
 
-        plt.imshow(aoi_mask, cmap='gray')
-        plt.title("Fixations per trial on the AOI mask")
-        plt.gca().set_ylim(plt.gca().get_ylim()[::-1]) # invert the y-axis to align with eyelink
-        plt.gca().xaxis.tick_top()
-        plt.scatter(x_coor, y_coor, color='skyblue', marker='x', s=15) # plot fixations within trial
-        plt.savefig(os.path.join(savepath, f'Fixations_in_AOI_{row_idx+1}.svg')) # save the svg for each trial in the savepath
+            plt.gca().set_ylim(plt.gca().get_ylim()[::-1]) # invert the y-axis to align with eyelink dataviewer origins and coordinates
 
-        # calculate ratio of fixations within the AOI for the current trial
-        fixations_within_aoi = 0 
+            plt.gca().xaxis.tick_top()
 
-        for x,y in zip(x_coor, y_coor):
-            if (x1 <= x <= x2) and (y1 <= y <= y2):
-                fixations_within_aoi += 1
-                ratio = 100* fixations_within_aoi/num_fixations
+            plt.scatter(epoch_x, epoch_y, color='skyblue', marker='o', s=15) # 
 
-        ratios_fixations_within.append(ratio)
+            if save_png and save_path:

```

Comment:
Users should be able to save the figure without providing a path – the path will just be the current path.